### PR TITLE
clarify thresholds for spike and flat_line

### PR DESCRIPTION
**Pull request by jessicaaustin**
Split variable 'thresholds' into 'suspect_threshold' and 'fail_threshold'.

This is based on user feedback -- one of the most common questions was "what does `thresholds` mean?". Of course they could read the docs, but I think it's helpful to make it as easy to understand as possible with just a quick overview.

Before:

selection_109

After:

selection_111

### CODE REVIEW

File: ioos_qc/qartod.py [hunk 1]
```diff
@@ -320,31 +320,28 @@ def climatology_test(config : Union[ClimatologyConfig, Sequence[Dict[str, Tuple]
 
 
 def spike_test(inp : Sequence[N],
-               thresholds : Tuple[N, N]
+               suspect_threshold: float,
+               fail_threshold: float
```
Comment (kwilcox) [inline ioos_qc/qartod.py]: Can you change these to `N` to document you can use a `Double` or really any numeric that can be compared to the `inp` using `>` and `<`?

File: tests/test_qartod.py [hunk 2]
```diff
@@ -622,19 +634,12 @@ def test_flat_line_missing_values(self):
...
                 inp=i,
                 tinp=self.times,
-                thresholds=self.thresholds,
+                suspect_threshold=self.suspect_threshold,
+                fail_threshold=self.fail_threshold,
                 tolerance=self.tolerance
             )
             npt.assert_array_equal(result, expected)
 
-    def test_flat_line_bad_input(self):
```
Comment (kwilcox) [inline tests/test_qartod.py]: Why did you need to remove this?

File: ioos_qc/qartod.py (hunk 1 above)
Comment (jessicaaustin) [reply ioos_qc/qartod.py]: Good catch. Updated to use `N`.

File: tests/test_qartod.py (hunk 2 above)
Comment (jessicaaustin) [reply tests/test_qartod.py]: The flat line thresholds are seconds instead of counts, so I don't think this is a valid test case anymore.
